src/presenters/open-graph/image-presenter.php

```diff
@@ -88,20 +88,56 @@ public function get() {
 	/**
 	 * Run the image content through the `wpseo_opengraph_image` filter.
 	 *
-	 * @param array $image The image.
+	 * @param array<string,int> $image The image.
 	 *
-	 * @return array The filtered image.
+	 * @return array<string,int> The filtered image.
 	 */
 	protected function filter( $image ) {
 		/**
-		 * Filter: 'wpseo_opengraph_image' - Allow changing the Open Graph image.
+		 * Filter: 'wpseo_opengraph_image' - Allow changing the Open Graph image url.
 		 *
 		 * @param string                 $image_url    The URL of the Open Graph image.
 		 * @param Indexable_Presentation $presentation The presentation of an indexable.
 		 */
-		$image_url = \trim( \apply_filters( 'wpseo_opengraph_image', $image['url'], $this->presentation ) );
+		$image_url = \apply_filters( 'wpseo_opengraph_image', $image['url'], $this->presentation );
 		if ( ! empty( $image_url ) && \is_string( $image_url ) ) {
-			$image['url'] = $image_url;
+			$image['url'] = \trim( $image_url );
+		}
+
+		$image_type = ( $image['type'] ?? null );
+		/**
+		 * Filter: 'wpseo_opengraph_image_type' - Allow changing the Open Graph image type.
+		 *
+		 * @param string                 $image_type   The type of the Open Graph image.
+		 * @param Indexable_Presentation $presentation The presentation of an indexable.
+		 */
+		$image_type = \apply_filters( 'wpseo_opengraph_image_type', $image_type, $this->presentation );
+		if ( ! empty( $image_type ) && \is_string( $image_type ) ) {
+			$image['type'] = \trim( $image_type );
+		}
+
+		$image_width = ( $image['width'] ?? null );
```

Comment:
I would make this into an empty string, instead of `null` to much the parameter type of the following filter signature

Reply:
Same for the height below and the type above.